pilot/pkg/model/sidecar.go

```diff
@@ -517,9 +518,11 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 }
 
 func matchingServices(importedHosts []host.Name, service *Service, ilw *IstioEgressListenerWrapper) []*Service {
-	// If a listener is defined with port, we should match services with port.
-	// If an unix domain socket is given as a port, we should not match by port and include services based on hosts.
-	needsPortMatch := ilw.IstioListener != nil && ilw.IstioListener.Port != nil && ilw.IstioListener.Port.GetNumber() != 0
+	// If a listener is defined with port, we should match services with port except in the following cases.
+	//  - If an unix domain socket is given as a port, we should not match by port and include services based on hosts.
```

Comment:
> If an unix domain socket is given as a port

How can i comprehend this?

Reply:
I think that comment is redundant. I will remove that - basically if a port has some number we should proceed with matching - does not matter whether it is a unix domain port or not.